src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -495,8 +495,11 @@ def delete_role_assignments(cmd, ids=None, assignee=None, role=None, resource_gr
     definitions_client = factory.role_definitions
     ids = ids or []
     if ids:
-        if assignee or role or resource_group_name or scope or include_inherited:
-            raise CLIError('When assignment ids are used, other parameter values are not required')
+        ids_override_args = ['assignee', 'role', 'resource_group_name', 'scope', 'include_inherited']
+        for arg in ids_override_args:
+            if locals()[arg]:
+                logger.warning("option '%s' will be ignored due to use of '--ids'.",
+                               cmd.arguments[arg].type.settings['options_list'][0])
```

Comment:
This mimics the behavior of core for `--ids` handling:

https://github.com/Azure/azure-cli/blob/ce3a1f1ddea0b59f0a5803bb427c55862ccae72b/src/azure-cli-core/azure/cli/core/commands/arm.py#L265-L268

A difference is that core doesn't show warning if default value is used, but we do here. I feel being explicit is better than being implicit.